boot-qemu.py

```diff
@@ -733,8 +744,58 @@ def launch_qemu(cfg):
     gdb_bin = cfg["gdb_bin"]
     kernel_location = cfg["kernel_location"]
     qemu_cmd = cfg["qemu_cmd"]
+    share_folder_with_guest = cfg["share_folder_with_guest"]
     timeout = cfg["timeout"]
 
+    if share_folder_with_guest and not interactive:
+        utils.yellow(
+            'Shared folder requested without an interactive session, ignoring...'
+        )
+        share_folder_with_guest = False
+    if share_folder_with_guest and gdb:
+        utils.yellow(
+            'Shared folder requested during a debugging session, ignoring...')
+        share_folder_with_guest = False
+
+    if share_folder_with_guest:
+        shared_folder.mkdir(exist_ok=True, parents=True)
+
+        # If shared folder was requested, we need to search for virtiofsd in
+        # certain known locations.
+        qemu_prefix = Path(qemu_cmd[0]).resolve().parent.parent
+        virtiofsd_locations = [
+            Path('libexec', 'virtiofsd'),  # Default QEMU installation, Fedora
+            Path('lib', 'qemu', 'virtiofsd'),  # Arch Linux, Debian, Ubuntu
+        ]
+        virtiofsd = utils.find_first_file(qemu_prefix, virtiofsd_locations)
+
+        if not (sudo := shutil.which('sudo')):
+            raise Exception(
+                'sudo is required to use virtiofsd but it could not be found!')
+        utils.green(
+            'Requesting sudo permission to run virtiofsd in the background...')
+        subprocess.run([sudo, 'true'], check=True)
```

Comment:
This runs `sudo true`? Why?

Reply:
To gain access to `sudo` for `virtiofsd`. We are running `virtiofsd` in the background so we cannot prompt for sudo then. `true` is a no-op command so we won't get any output beyond the prompt.

Reply:
I think a detailed comment would go a long way here.

Reply:
I added some more information about this to the docstring for `get_and_call_sudo()`.